backend/open_api_spec.yml

```diff
@@ -0,0 +1,829 @@
+        one does not exist, then delete the Dataset from the revision.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/optional_query_curator_tag"
+        - $ref: "#/components/parameters/optional_query_dataset_uuid"
+        - $ref: "#/components/parameters/query_visibility"
+      responses:
+        "204":
+          $ref: "#/components/responses/204"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+    get:
+      description: Fetch a Dataset.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/optional_query_curator_tag"
+        - $ref: "#/components/parameters/optional_query_dataset_uuid"
+        - $ref: "#/components/parameters/query_visibility"
+      responses:
+        "200":
+          description: OK
+            content:
+              application/json:
+                schema:
+                  type: object
+                  properties:
+                    dataset:
+                      $ref: "#/components/schemas/dataset"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+    patch:
+      description: Update a Dataset's curator-provided tag.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/optional_query_curator_tag"
+        - $ref: "#components/parameters/optional_query_dataset_uuid"
+        - $ref: "#/components/parameters/query_visibility"
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              additionalProperties: false
+              properties:
+                curator_tag:
+                  type: string
+                  description: curator-provided tag
+      responses:
+        "204":
+          $ref: "#/components/responses/204"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+
+  /curation/collections/{collection_id}/datasets/upload-links:
+    put:
+      description: Upload a Dataset. The Dataset will be downloaded from the provided link.
+      security:
+        - cxguserCookie: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              additionalProperties: false
+              properties:
+                curator_tag:
+                  type: string
+                id:
+                  $ref: "#/components/schemas/dataset_uuid"
+                link:
+                  $ref: "#/components/schemas/upload_link"
+      responses:
+        "204":
```

Comment:
return 202 since this an async operation

Reply:
I know I assented to this when we were meeting earlier today, but I disagree after thinking about it more. The actions taken by the server have been completed; there is no queued work. The fact that the next thing for the curator to do is to upload to s3 does not impinge on whether the response from this endpoint is considered to be asynchronous at all.